test/correctness/math.cpp

```diff
@@ -4,6 +4,14 @@
 #include <iostream>
 #include <limits>
 
+#if defined(_MSC_VER) && (_MSC_VER < 1700)
+// These functions don't exist in msvc < 2012 (i.e., when _MSC_VER < 1700)
+#define asinh(x)    (log(x + sqrt(x*x + 1)))
```

Comment:
Please prefer inline functions to macros.

Reply:
Halide no longer supports any MSVC earlier than 2015.